packages/salesforce-adapter/src/filters/email_template_static_files.ts

```diff
@@ -65,25 +65,37 @@ const findFolderPath = (instance: InstanceElement): string =>
   `${SALESFORCE}/${RECORDS_PATH}/${EMAIL_TEMPLATE_METADATA_TYPE}/${instance.value.fullName}`
 
 const organizeStaticFiles = async (instance: InstanceElement): Promise<void> => {
-  const folderPath = findFolderPath(instance)
-  if (_.isUndefined(folderPath)) {
-    const instApiName = await apiName(instance)
-    log.warn(`could not extract the attachments of instance ${instApiName}, instance path is undefined`)
-  } else {
-    const emailName = `${folderPath.split('/').pop()}.email`
-    instance.value.content = new StaticFile({
-      filepath: path.join(folderPath, emailName),
-      content: await instance.value.content.getContent(),
-    })
-    instance.value.attachments = makeArray(instance.value.attachments)
-    if (isEmailAttachmentsArray(instance.value)) {
-      instance.value.attachments.forEach(attachment => {
-        attachment.content = createStaticFile(
-          // attachmen.content type is a string before the creation of the static file
-          folderPath, attachment.name, attachment.content as string
-        )
+  try {
+    const folderPath = findFolderPath(instance)
+    if (_.isUndefined(folderPath)) {
+      const instApiName = await apiName(instance)
+      log.warn(`could not extract the attachments of instance ${instApiName}, instance path is undefined`)
+    } else {
+      const emailName = `${folderPath.split('/').pop()}.email`
+      instance.value.content = new StaticFile({
+        filepath: path.join(folderPath, emailName),
+        content: await instance.value.content.getContent(),
       })
+      instance.value.attachments = makeArray(instance.value.attachments)
+      if (isEmailAttachmentsArray(instance.value)) {
+        instance.value.attachments.forEach(attachment => {
+          attachment.content = createStaticFile(
+            // attachment.content type is a string before the creation of the static file
+            folderPath, attachment.name, attachment.content as string
+          )
+        })
+      } else if (!Array.isArray(instance.value.attachments)) {
+        log.debug('email template attachments are not an array: %s',
+          safeJsonStringify(instance.value.attachments,
+            (_key, value) => (_.isString(value) ? _.truncate(value) : value)))
```

Comment:
probably would be better to use `inspect` and not `safeJsonStringify` here, we've seen performance issues in other adapters.
#4267 hasn't been merged yet but some places were already changed after they "blew up" in certain scenarios

I think you don't even have to call it directly, could just replace this with
```suggestion
        log.debug('email template attachments are not an array: %o', instance.value.attachments)
```

Reply:
Won't this end up logging the entire contents, which may be several MB in size?

Reply:
exactly the opposite
inspect has a bunch of safeties in its options (max depth, max string length, max keys, etc) that `safeJsonStringify` doesn't

Reply:
Makes sense. However, I tried using `inspect` with the `maxStringLength` option, but that option was added in Node.js v12.17 and we currently use `@types/node` v12.7.1 so the code fails to compile. Is it worth upgrading `@types/node` in our project for this change?

Reply:
you don't need to call `inspect` explicitly
`%o` does that for you and the defaults are good enough

Reply:
Docs say the default is no limit on string length though?

> maxStringLength <integer> Specifies the maximum number of characters to include when formatting. Set to null or Infinity to show all elements. Set to 0 or negative to show no characters. Default: Infinity.

Reply:
huh, you are right, my bad
I think we can change our node types to at least the latest 12.x.x
hopefully soon we'll be able to bump it beyond that

Reply:
I tried and got some type errors in jsforce-types. I'm thinking it's not worth it.